ResourceProvisioner/src/ResourceProvisioner.Infrastructure/Services/RepositoryService.cs

```diff
@@ -80,7 +80,17 @@ public async Task<PullRequestUpdateMessage> HandleResourcing(CreateResourceRunCo
         };
 
         _semaphore.Release();
```

Comment:
will semphore deadlock if the code inbetween the async wait and the release causes a fault ?

Reply:
No need as it's a single lived run and exceptions are handled already. Removed it as it's not needed